polkadot/node/collation-generation/src/lib.rs

```diff
@@ -178,95 +172,130 @@ impl CollationGenerationSubsystem {
+
+		let session_info = self.session_info.get(relay_parent, session_index, ctx.sender()).await?;
+		let collation = PreparedCollation {
+			collation,
+			relay_parent,
+			para_id: config.para_id,
+			validation_data,
+			validation_code_hash,
+			n_validators: session_info.n_validators,
+			core_index,
+			session_index,
+		};
+
+		construct_and_distribute_receipt(
+			collation,
+			config.key.clone(),
+			ctx.sender(),
+			result_sender,
+			&mut self.metrics,
+			session_info.v2_receipts,
+		)
+		.await?;
+
+		Ok(())
 	}
 
-	let para_id = config.para_id;
+	async fn handle_new_activation<Context>(
+		&mut self,
+		activated: impl IntoIterator<Item = Hash>,
+		ctx: &mut Context,
+	) -> Result<()> {
+		let Some(config) = &self.config else {
+			return Ok(());
+		};
 
-	let _overall_timer = metrics.time_new_activations();
+		// If there is no collation function provided, bail out early.
+		// Important: Lookahead collator and slot based collator do not use `CollatorFn`.
+		if config.collator.is_none() {
+			return Ok(())
+		}
 
-	for relay_parent in activated {
-		let _relay_parent_timer = metrics.time_new_activations_relay_parent();
+		let para_id = config.para_id;
 
-		let (availability_cores, validators, async_backing_params) = join!(
-			request_availability_cores(relay_parent, ctx.sender()).await,
-			request_validators(relay_parent, ctx.sender()).await,
-			request_async_backing_params(relay_parent, ctx.sender()).await,
-		);
+		let _overall_timer = self.metrics.time_new_activations();
 
-		let availability_cores = availability_cores??;
-		let async_backing_params = async_backing_params?.ok();
-		let n_validators = validators??.len();
-		let maybe_claim_queue = fetch_claim_queue(ctx.sender(), relay_parent)
-			.await
-			.map_err(crate::error::Error::UtilRuntime)?;
-
-		// The loop bellow will fill in cores that the para is allowed to build on.
-		let mut cores_to_build_on = Vec::new();
-
-		// This assumption refers to all cores of the parachain, taking elastic scaling
-		// into account.
-		let mut para_assumption = None;
-		for (core_idx, core) in availability_cores.into_iter().enumerate() {
-			// This nested assumption refers only to the core being iterated.
-			let (core_assumption, scheduled_core) = match core {
-				CoreState::Scheduled(scheduled_core) =>
-					(OccupiedCoreAssumption::Free, scheduled_core),
-				CoreState::Occupied(occupied_core) => match async_backing_params {
-					Some(params) if params.max_candidate_depth >= 1 => {
+		for relay_parent in activated {
+			let _relay_parent_timer = self.metrics.time_new_activations_relay_parent();
+
+			let session_index =
+				request_session_index_for_child(relay_parent, ctx.sender()).await.await??;
+
+			let session_info =
+				self.session_info.get(relay_parent, session_index, ctx.sender()).await?;
+			let async_backing_params = session_info.async_backing_params;
+			let n_validators = session_info.n_validators;
+
+			let availability_cores =
```

Comment:
Do we really need the availability cores if we have the claim queue and backing state ?

Reply:
The only reason would be to preserve synchronous backing. but considering that we are removing the async backing params I think we can remove this.

This test will probably start failing: https://github.com/paritytech/polkadot-sdk/blob/9584dbda9ed5fc91b1837f35f401659239fdaff1/polkadot/zombienet_tests/functional/0017-sync-backing.toml#L9 sometime in the future. We should fix then it by adding a test parachain that produces 12-second blocks.

Actually, I think I can remove the backing state runtime API call also.

Reply:
done